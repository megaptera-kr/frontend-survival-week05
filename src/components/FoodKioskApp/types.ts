export type CategoryType = '전체' | '중식' | '한식' | '일식';

export type MenuItem = {
  id: string;
  name: string;
  price: number;
};

export type ReceiptType = {
  id?: string;
  menu?: MenuItem[];
  totalPrice?: number;
};

export type ReceiptResponse = {
  receipt: ReceiptType;
};

export type RestaurantItem = {
  id: string;
  category: string;
  name: string;
  menu: MenuItem[];
};

export type RestaurantsItem = {
  restaurants: RestaurantItem[];
};

export type RenderType = 'none' | 'card' | 'notFound';

export interface State {
  restaurants: RestaurantItem[];
  category: CategoryType;
  query: string;
  receipt: ReceiptType;
  addedMenus: MenuItem[];
}

export type Action =
  | { type: 'fetch_data'; payload: RestaurantItem[] }
  | { type: 'update_category'; payload: CategoryType }
  | { type: 'search'; payload: string }
  | { type: 'add_menu'; payload: MenuItem }
  | { type: 'delete_menu'; payload: number }
  | { type: 'reset_menu'; payload: MenuItem[] }
  | { type: 'get_receipt'; payload: ReceiptType }
  | { type: 'delete_receipt'; payload: ReceiptType };
