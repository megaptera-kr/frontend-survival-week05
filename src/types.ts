type Category = '한식' | '중식' | '일식';

export type Menu = {
  id: string;
  name: string;
  price: number;
}

export type Restaurants = {
  id: string;
  category: Category;
  name: string;
  menu: Menu[]
}

export type Data = {
  restaurants: Restaurants[];
};

export type Receipt = {
  id: string
  menu: Menu[]
  totalPrice: number
}

export type ReceiptData = {
  receipt: Receipt;
};
