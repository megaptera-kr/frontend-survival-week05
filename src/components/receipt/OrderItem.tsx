import Food from '../../types/Food';

type OrderItemProps = {
  food: Food;
}

function OrderItem({ food }: OrderItemProps) {
  const { name, price } = food;
  return (
    <li
      style={{
        display: 'flex',
        paddingBlock: '.5rem',
      }}
    >
      <span style={{ margin: '0 auto' }}>
        {name}
        (
        {price.toLocaleString()}
        Ïõê)
      </span>
    </li>
  );
}

export default OrderItem;
