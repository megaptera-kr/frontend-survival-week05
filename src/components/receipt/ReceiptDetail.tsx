import Food from '../../types/food';
import Receipt from '../../types/receipt';

type ReceiptDetailProps = {
  receipt: Receipt
}

export default function ReceiptDetail({ receipt }: ReceiptDetailProps) {
  return (
    <div>
      <h3>영수증</h3>
      <p>
        {`주문번호: ${receipt.id}`}
      </p>
      <ul>
        {receipt.menu.map((food: Food, index: number) => (
          <li
            // eslint-disable-next-line react/no-array-index-key
            key={`${food.id}-${index}`}
            style={{
              display: 'flex',
              justifyContent: 'space-between',
              paddingBottom: '10px',
            }}
          >
            <p>
              {`${food.name}(${food.price.toLocaleString('ko-kr')}원)`}
            </p>
          </li>
        ))}
      </ul>
      <p />
      <p>
        {`총 가격: ${receipt.totalPrice.toLocaleString('ko-kr')}원`}
      </p>
    </div>
  );
}
