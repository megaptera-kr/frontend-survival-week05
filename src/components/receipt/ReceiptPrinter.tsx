import _ from 'lodash';

import Receipt from '../../types/Receipt';
import MenuItem from '../restaurants/MenuItem';

type ReceiptPrinter = {
  receipt: Receipt;
}

export default function ReceiptPrinter({ receipt }: ReceiptPrinter) {
  if (_.isEmpty(receipt)) {
    return <p>[영수증 나오는 곳]</p>;
  }

  const { id, menu, totalPrice } = receipt;

  return (
    <div>
      <h2>영수증</h2>
      <h3>주문번호</h3>
      <p>{id}</p>
      <h3>주문목록</h3>
      {menu.map((food, index) => {
        const key = `${food.id}-${index}`;

        return (
          <MenuItem
            key={key}
            food={food}
          />
        );
      })}
      <p>
        총 가격:
        {' '}
        {`${totalPrice.toLocaleString()}원`}
      </p>
    </div>
  );
}
