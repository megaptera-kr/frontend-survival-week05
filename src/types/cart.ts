import type { Food } from './food';

export interface OrderRequest {
  menu: Food[];
  totalPrice: string;
}
