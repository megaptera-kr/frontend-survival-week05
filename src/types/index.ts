import { Dispatch, SetStateAction } from 'react';

export type Menu = {
    id : string,
    name : string,
    price : number
}

export type Restaurant = {
    id : string,
    category : string,
    name : string,
    menu : Menu[]
}

export type Receipt = {
    id : string,
    menu : Menu[],
    totalPrice : number
}

export type Category = '전체' | '중식' | '한식' | '일식';

export type SetValue<T> = Dispatch<SetStateAction<T>>
