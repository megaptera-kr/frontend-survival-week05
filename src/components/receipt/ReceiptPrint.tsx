import Receipt from '../../types/receipt';
import ReceiptDetail from './ReceiptDetail';

type ReceiptPrintProps = {
  receipt: Receipt
}

export default function ReceiptPrint({ receipt }:ReceiptPrintProps) {
  return (
    <div>
      <p>[영수증 나오는 곳]</p>
      { receipt.id
        && (
          <ReceiptDetail receipt={receipt} />
        )}
    </div>
  );
}
