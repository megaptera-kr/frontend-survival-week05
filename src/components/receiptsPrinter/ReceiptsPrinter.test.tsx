import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import ReceiptsPrinter from './ReceipsPrinter';
import { Receipts } from '../../types/receipt';
import { receipt } from '../../../fixtures';

const context = describe;

describe('RecieptsPrinter', () => {
  function renderReceiptsPrinter(receipts?: Receipts) {
    return render(<ReceiptsPrinter receipts={receipts} />);
  }

  it('render correctly', () => {
    const component = renderer.create(<ReceiptsPrinter receipts={receipt} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  context('when dont have receipts', () => {
    it('display [영수증 나오는 곳]', () => {
      renderReceiptsPrinter();

      expect(screen.getByText('[영수증 나오는 곳]')).toBeInTheDocument();
    });
  });

  context('when have receipts', () => {
    it('display totlaPrice', () => {
      renderReceiptsPrinter(receipt);

      expect(screen.getByText('총 가격: 13,000원')).toBeInTheDocument();
    });
  });
});
