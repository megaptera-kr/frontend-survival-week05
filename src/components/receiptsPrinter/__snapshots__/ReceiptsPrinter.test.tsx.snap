// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecieptsPrinter render correctly 1`] = `
<article>
  <h2>
    영수증
  </h2>
  <table
    style={
      {
        "backgroundColor": "black",
        "border": "1px solid black",
      }
    }
  >
    <caption>
      영수증
    </caption>
    <thead
      style={
        {
          "backgroundColor": "white",
        }
      }
    >
      <tr>
        <th
          scope="col"
        >
          주문번호
        </th>
        <th
          scope="col"
        >
          주문목록
        </th>
        <th
          scope="col"
        >
          총 가격
        </th>
      </tr>
    </thead>
    <tbody
      style={
        {
          "backgroundColor": "white",
        }
      }
    >
      <tr>
        <td>
          RECEIPT_ID
        </td>
        <td>
          <ul
            style={
              {
                "display": "flex",
                "flexDirection": "column",
                "gap": "4px",
                "listStyle": "none",
                "padding": "0",
              }
            }
          >
            <li>
              <div
                style={
                  {
                    "columnGap": "4px",
                    "display": "flex",
                    "justifyContent": "space-between",
                  }
                }
              >
                <div
                  style={
                    {
                      "display": "flex",
                      "flexDirection": "column",
                    }
                  }
                >
                  <span>
                    이름: 짜장면
                  </span>
                  <span>
                    가격: 8,000
                  </span>
                </div>
              </div>
            </li>
            <li>
              <div
                style={
                  {
                    "columnGap": "4px",
                    "display": "flex",
                    "justifyContent": "space-between",
                  }
                }
              >
                <div
                  style={
                    {
                      "display": "flex",
                      "flexDirection": "column",
                    }
                  }
                >
                  <span>
                    이름: 짬뽕
                  </span>
                  <span>
                    가격: 5,000
                  </span>
                </div>
              </div>
            </li>
          </ul>
        </td>
        <td>
          총 가격: 13,000원
        </td>
      </tr>
    </tbody>
  </table>
</article>
`;
